src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx

```diff
@@ -402,7 +402,7 @@ function ScatterplotViz(props: VisualizationProps<Options>) {
           ? vizConfig.independentAxisValueSpec
           : 'Full',
         dependentAxisValueSpec: 'Full',
-        markerBodyOpacity: 0,
+        markerBodyOpacity: 0.5,
```

Comment:
Looks like we're still setting `markerBodyOpacity` to the now-default value (0.5) whenever the `vizConfig` changes.

Reply:
Good question @jernestmyers! That part is related to the change of input variable (X, Y, etc.). We usually reset most of plot control related ones to its default whenever input variable is changed: e.g., just above the line, axis range option of dependent axis is reset to its default, 'Full' when a variable is changed. Due to this reason, I followed that rule for marker opacity as well. Does this make sense to you?

Reply:
@jernestmyers Oh... I missed that part " i think if the user changes the value of the slider, it should be remembered when changing variables rather than going back to default.". When I said "remembering user's choice of opacity" in my summary, I meant a case when a user goes out of Viz and comes back. Good catch! 👍 I will change it accordingly.

Reply:
@jernestmyers I reflected it. The line can be simply removed to preserve the marker opacity value when changing an input variable 👍